### PR TITLE
fix: closed stream

### LINKED ISSUE #4: Some adapters will close the stream themselves
When using the wrapper with https://github.com/thephpleague/flysystem-azure-blob-storage, the underlying adapter will take over handling the passed stream handle. It means that `$this->current->handle` could be closed when trying to `fseek()` and close it.

### PR DESCRIPTION
**Pull request by dkarlovi**
Closes #4.

### CODE REVIEW

**Comment by m2mtech** [comment]
Dear @dkarlovi
this change breaks a unit test - which seems ok for me at this moment - could you please adjust the unit test?
Thanks for your contribution!

**Comment by dkarlovi** [comment]
> could you please adjust the unit test?

Sure. :+1: I'll check what exactly broke.

File: src/Flysystem/StreamWrapper.php [hunk 1]
```diff
@@ -106,7 +101,7 @@ public function stream_flush(): bool
                 $success = false;
             }
 
-            if (false !== $currentPosition) {
+            if (false !== $currentPosition && is_resource($this->current->handle)) {
```
Comment (m2mtech) [inline src/Flysystem/StreamWrapper.php:104]: If the handle is not a resource it would have triggered an exception above.

Comment (dkarlovi) [reply src/Flysystem/StreamWrapper.php:104]: It wouldn't, the `writeStream()` call closes the resource. Once it reaches here, it's a closed resource, which doesn't pass `is_resource()` (same issue as with the `stream_close()` above complaining when you pass it a closed resource to close).

Comment (m2mtech) [reply src/Flysystem/StreamWrapper.php:104]: Could you please provide a test case where this is triggered?

**Comment by dkarlovi** [comment]
@m2mtech sure, I'll test it out and let you know. My assumption is the fact `fseek()` is being done on a closed resource will be the issue here still, but will check. The changes made in this and the next PR were made with the app having everything set up with Azure Blob Storage (via Flysystem) and it 100% didn't work before the changes, but 100% works after them.

**Pull request closed without merging**